src/server_connection.rs

```diff
@@ -0,0 +1,120 @@
+use reqwest::{Client, Error, IntoUrl, Response, Url};
+use serde::de::DeserializeOwned;
+use serde::Serialize;
+
+/// Provides basic REST communication with the server.
+pub struct ServerConnection {
+    /// The main url without any routes.
+    base_url: Url,
+    /// HTTP client, uses an internal connection pool and can thus be shared
+    http_client: Client,
+}
+
+impl ServerConnection {
+    /// Constructor for creating a new Server Connection.
+    /// # Arguments
+    /// * 'base_url' the url of the server
+    pub fn new<U: IntoUrl>(base_url: U) -> Self {
+        let url = base_url.into_url().unwrap();
+        assert!(url.path().ends_with('/'), "base_url path must end with '/'");
+        Self {
+            base_url: url,
+            http_client: Client::new(),
+        }
+    }
+
+    pub fn base_url(&self) -> &Url {
+        &self.base_url
+    }
```

Comment:
@iTitus  could you add documentation on public functions :wink:

Reply:
this method does literally nothing